src/components/wrappers/ProcessWrapper.tsx

```diff
@@ -40,10 +40,9 @@ export const ProcessWrapper = ({ isFetching }: IProcessWrapperProps) => {
   const instanceIdFromUrl = useInstanceIdParams()?.instanceId;
   window.instanceId = instanceIdFromUrl;
 
-  const { pdfPreview } = useAppSelector((state) => state.devTools);
   const [searchParams] = useSearchParams();
   const renderPDF = searchParams.get('pdf') === '1';
-  const previewPDF = searchParams.get('pdf') === 'preview' || pdfPreview;
+  const { pdfPreview: previewPDF } = useAppSelector((state) => state.devTools);
```

Comment:
This would trigger a re-render every time anything in the devTools state changes. It's better to just select the property you need to read.

```suggestion
  const previewPDF = useAppSelector((state) => state.devTools.pdfPreview);
```

Reply:
Right, stupid 🤦